### ISSUE TITLE
Style/IfWithSemicolon incorrectly auto corrects and flips conditional value

### ISSUE THREAD

**Issue by willhoney7** [rubocop/rubocop#15034]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Incorrect autocorrect

### Cop name

Style/IfWithSemicolon

### Steps to reproduce the problem

With just the "Style/IfWithSemicolon" rule enabled (see rubocop below), 

Given the following file:
```ruby
class FormatTest
  def truthy
    true
  end

  def run
    unless truthy; fail("should not error"); end
  end
end

FormatTest.new.run
```
Then in the command line...
```bash
# execute the file
$ ruby ./bug.rb 
    # no error

# correct the issue
$ rubocop -a ./bug.rb

# now triggers the error
$ ruby ./bug.rb 
bug.rb:7:in 'FormatTest#run': should not error (RuntimeError)
	from bug.rb:11:in '<main>'
```
Formatted output
```ruby
class FormatTest
  def truthy
    true
  end

  def run
    truthy ? fail("should not error") : nil
  end
end

FormatTest.new.run
```

### Expected behavior

The code to be formatted correctly without reversing the conditional logic.

### Actual behavior

The code is updated and changes the actual logic.

### RuboCop configuration

```yaml
AllCops:
  DisabledByDefault: true

Style/IfWithSemicolon:
  Enabled: true
```

### RuboCop version

```shell
1.85.1 (using Parser 3.3.10.2, rubocop-ast 1.49.1, analyzing as Ruby 2.7, running on ruby 3.3.0) [arm64-darwin23]
```
